sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Common/MetricsTest.cs

```diff
@@ -422,7 +422,9 @@ public void Complexity()
             Complexity(AnalyzerLanguage.CSharp, "class MyClass { void MyMethod() { var t = null ?? 0; } }")
                 .Should().Be(2);
             Complexity(AnalyzerLanguage.CSharp, "class MyClass { void MyMethod(int? t) { t ??= 0; } }")
-                .Should().Be(1); // null-coalescing assignment is not considered when computing cyclomatic complexity. See: https://github.com/SonarSource/sonar-dotnet/issues/3518
+                .Should().Be(2);
```

Comment:
from the [discussion with Ann](https://discuss.sonarsource.com/t/cognitive-complexity-for-null-coalesce-and-null-coalesce-assignment-operators/5432/2), I understand that we shouldn't treat this differently 

Although chaining these assignments is still a bit messy. I'll wait for the test to be updated.

Reply:
From what I understood the discussion was related to cognitive complexity and this issue is related to cognitive complexity which is a different topic.

When computing cyclomatic complexity, `?.` is used like:
```
            var x = c?.Name?.ToString();
//                   ^ Secondary [10] {{+1}}
//                         ^ Secondary@-1 [10] {{+1}}
```